From 0ddc7727381d69eb54a17fa27114554712f6f5d5 Mon Sep 17 00:00:00 2001
From: Ryan Duffin <rad.duffin@gmail.com>
Date: Tue, 25 Jul 2017 17:17:29 -0400
Subject: [PATCH] Change GraphML namespace fprintf calls to use format
 specifiers (#1020)

---
 src/foreign-graphml.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/foreign-graphml.c b/src/foreign-graphml.c
index 1df2ec8f..13dbced1 100644
--- a/src/foreign-graphml.c
+++ b/src/foreign-graphml.c
@@ -1405,13 +1405,13 @@ int igraph_write_graph_graphml(const igraph_t *graph, FILE *outstream,
   
   ret=fprintf(outstream, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n");
   if (ret<0) IGRAPH_ERROR("Write failed", IGRAPH_EFILE);
-  ret=fprintf(outstream, "<graphml xmlns=\"" GRAPHML_NAMESPACE_URI "\"\n");
+  ret=fprintf(outstream, "<graphml xmlns=\"%s\"\n", GRAPHML_NAMESPACE_URI);
   if (ret<0) IGRAPH_ERROR("Write failed", IGRAPH_EFILE);
   ret=fprintf(outstream, "         xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n");
   if (ret<0) IGRAPH_ERROR("Write failed", IGRAPH_EFILE);
-  ret=fprintf(outstream, "         xsi:schemaLocation=\"" GRAPHML_NAMESPACE_URI "\n");
+  ret=fprintf(outstream, "         xsi:schemaLocation=\"%s\n", GRAPHML_NAMESPACE_URI);
   if (ret<0) IGRAPH_ERROR("Write failed", IGRAPH_EFILE);
-  ret=fprintf(outstream, "         " GRAPHML_NAMESPACE_URI "/1.0/graphml.xsd\">\n");
+  ret=fprintf(outstream, "         %s/1.0/graphml.xsd\">\n", GRAPHML_NAMESPACE_URI);
   if (ret<0) IGRAPH_ERROR("Write failed", IGRAPH_EFILE);
   ret=fprintf(outstream, "<!-- Created by igraph -->\n");
   if (ret<0) IGRAPH_ERROR("Write failed", IGRAPH_EFILE);
-- 
2.13.6

